odoo/addons/test_lint/tests/test_checkers.py

```diff
@@ -89,3 +106,148 @@ def do_the_thing(self):
             self.env.cr.execute(f'select name from {self._table}')
         """)
         self.assertFalse(r, f'underscore-attributes are allowable\n{errs}')
+
+
+    @contextmanager
+    def assertMessages(self, *messages):
```

Comment:
`assertMessages()` with no parameters feels like it's asserting that *at least one message* is being generated, that's weird.
If we required the argument to be packed into a list, it would be more explicit, no?

```py
# no message, but explicit
assertMessages([])

# expected messages
assertMessages(["sql-injection"])
# or with a shorthand when the argument is a single msg? 
assertMessages("sql-injection")
```

other option, reintroduce `assertNoMessage()` for this case

Reply:
Another solution would also be to have the context manager only returning the messages
```
with linter.catchMesages as messages:
    call()
    self.assertEqual(messages, [])
```
This would avoid to have either the test case need the linter or the linter need the test case.
Not sure what's the best